cfg/config.go

```diff
@@ -266,18 +266,24 @@ func Parse(ptr interface{}, values map[string]int) error {
 // 	}
 // }
 // this function will set `config.Child.T = config.T`
-// see file.d/cfg/config_test.go:TestHierarchy for an example
+// see file.d/cfg/config_test.go:TestHierarchy for an example.
 func ParseChild(parent reflect.Value, v reflect.Value, values map[string]int) error {
 	if v.CanAddr() {
-		for i := 0; i < v.NumField(); i++ {
-			name := v.Type().Field(i).Name
+		// Check for cfg pointers.
+		vIndirect := reflect.Indirect(v)
```

Comment:
Need add new unit test, which shows a problem

Reply:
It's optional in this case.